assay/src/org/labkey/assay/plate/AssayPlateMetadataServiceImpl.java

```diff
@@ -1045,6 +1054,64 @@ private void insertOrUpdateReplicateStats(
         }
     }
 
+    @Nullable
+    private static DataState getStateFromRow(Container container, Map<String, Object> row, @Nullable DomainProperty stateProp) throws ExperimentException
+    {
+        if (stateProp != null)
+        {
+            Set<String> importAlias = new CaseInsensitiveHashSet(stateProp.getName());
+            importAlias.addAll(stateProp.getImportAliasSet());
+            for (Map.Entry<String, Object> entry : row.entrySet())
+            {
+                if (importAlias.contains(entry.getKey()))
+                {
+                    if (entry.getValue() instanceof Integer stateRowId)
+                    {
+                        DataState state = PlateDataStateManager.get().getStateForRowId(container, stateRowId);
+                        if (state == null)
+                            throw new ExperimentException(String.format("No data states for the rowID : %d was found", stateRowId));
+
+                        return state;
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
+    @Nullable
+    public static DomainProperty getAssayStateProp(Domain resultDomain)
+    {
+        if (resultDomain == null)
+            return null;
+
+        return resultDomain.getProperties().stream().filter(dp -> AssayResultDomainKind.STATE_COLUMN_NAME.equalsIgnoreCase(dp.getName()))
+                .findFirst().orElse(null);
+    }
+
+    /**
+     * Ensure the data state value in the row represents a data state in scope and is a valid state for plate based
+     * assays.
+     */
+    public static void validateRowDataStates(Container container, Map<String, Object> row, DomainProperty stateProp) throws ValidationException
+    {
+        try
+        {
+            DataState state = getStateFromRow(container, row, stateProp);
+            if (state != null)
+            {
+                if (PlateDataStateManager.StateType.getType(state.getStateType()) == null)
+                {
+                    throw new ValidationException(String.format("The data state : %s is not valid for this assay", state.getStateType()));
```

Comment:
I think we want to use `state.getLabel()` in the error message instead of `state.getStateType()`.

Also, the format of the error message seems odd to me with the colon in it. I'd suggest: 
```suggestion
                    throw new ValidationException(String.format("The data state '%s' is not valid for this assay.", state.getLabel()));
```